core/src/main/java/org/infinispan/configuration/cache/IndexingConfigurationBuilder.java

```diff
@@ -320,9 +320,9 @@ public IndexingConfiguration create() {
       // check for presence of index providers that are not persistent upon restart
       boolean isVolatile = typedProperties.entrySet().stream()
                                      .anyMatch(e -> {
-                                        if (((String) e.getKey()).endsWith(DIRECTORY_PROVIDER_SUFFIX)) {
+                                        if (((String) e.getKey()).endsWith("directory.type")) {
                                            String directoryImplementationName = String.valueOf(e.getValue()).trim();
-                                           return LOCAL_HEAP_DIRECTORY_PROVIDER.equalsIgnoreCase(directoryImplementationName)
+                                           return "local-heap".equalsIgnoreCase(directoryImplementationName)
```

Comment:
Why this change ? LOCAL_HEAP_DIRECTORY_PROVIDER is "local-heap"

Reply:
Ideally I would have not wanted to change the core at all. This is (and the subsequent comment is about this same change) the only change we did to the ISPN core. The reason is that `isVolatile` must be true if and only if the local heap is used as directory type and we have different names in Search 6 for such property (key and value). I don't know actually if there is any other way to reach the same goal.

Reply:
Sorry I didn't see well the constant value. Yes I think we can use it. I'm changing it...